### PR TITLE
Added XKCD command

### PR DESCRIPTION
**Pull request by romanpeters**
Returns the URL of the latest XKCD comic.

### CODE REVIEW

**Comment by RobinSikkens** [comment]
According to Bandit Rule B310 is blacklisted because of the following reason:

> Blacklist various Python calls known to be dangerous
> 
> This blacklist data checks for a number of Python calls known to have possible security implications. The following blacklist tests are run against any function calls encoutered in the scanned code base, triggered by encoutering ast.Call nodes.

You may have to consider using another method of access.

**Comment by RobinSikkens** [comment]
Discord.py reccommends the use of aiohttp (https://aiohttp.readthedocs.io/en/stable/) for http call as it works wel with async.

Example is this function from https://github.com/RobinSikkens/Sticky-discord/blob/master/stickord/helpers/xkcd_api.py:
```python
async def get_recent():
    ''' Get and parse the most recent xkcd comic. '''
    async with aiohttp.get('https://xkcd.com/info.0.json') as r:
return await r.json()
```

**Comment by romanpeters** [comment]
```RuntimeWarning: coroutine 'get_text' was never awaited```

I think it should be awaited in `api_requests.py` #8, right?

**Comment by RobinSikkens** [comment]
As `get_text()` and `get_json()` are async functions they have to be awaited. 
To get text from `get_text()` please use:
```python
text = await get_text(url)
```

**Comment by romanpeters** [comment]
I tried that, but then it returns the generator object itself.

**Comment by RobinSikkens** [comment]
I will look into it.
Furthermore i saw `api_requests.get_text(url, ("data", "img"))` in your code.
please note this will send (https://xkcd.com/info.0.json?data=img) as a request.

in order to get the desired effect you might want to use 
```python
xkcd = get_json(url)
return xkcd['img']
```

**Comment by romanpeters** [comment]
Ah right, I misinterpreted the function, I'll change it.

**Comment by RobinSikkens** [comment]
I fixed the generator problem in the latest commit of #8

**Comment by romanpeters** [comment]
It's working.
Would it be an improvement to use the `FileWrapper` to send an image instead of a URL? If so, I'll add that before merging.

Edit: 
I'll use discord.Embed

**Comment by romanpeters** [comment]
I've added a xkcd by number, search by text and random feature to the command.
The search function relies on the database. Comics retrieved by means other than the search function, get added to the database automatically.
It is also possible to add all comics to the database by running the script `scrips/xkcd_crawler.py`.

File: devbot/commands/xkcd.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+def embed(title, image_url) -> discord.Embed:
+    result = discord.Embed()
+    result.title = title
+    result.set_image(url=image_url)
+    return result
+
+
+@Command(["xkcd"])
+async def xkcd(message_contents, *_args, **_kwargs) -> (str, discord.Embed):
```
Comment (RobinSikkens) [inline devbot/commands/xkcd.py]: Here you give as return type (str, discord.Embed) whereas the returns return str or discord.Embed.

File: devbot/commands/xkcd.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    """ Returns a xkcd comic """
+    url = "https://xkcd.com/info.0.json"  # latest xkcd
+
+    if message_contents:
+        if len(message_contents) == 1:
```
Comment (RobinSikkens) [inline devbot/commands/xkcd.py:24]: You don't actually have an if statement for the use of `!xkcd` without arguments. As it functions now it wil always return the first ever xkcd. Maybe have it return the latest instead?

File: scripts/xkcd_crawler.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
+"""
+Retrieve all xkcd meta data
+Expected runtime 3.5 minutes
+
+Requierment: requests
+Optional: tqdm
+"""
+import json
+import devbot.database as db
+from devbot.tools import api_requests
```
Comment (RobinSikkens) [inline scripts/xkcd_crawler.py]: As the requests are handled by requests this import is redundant.

File: scripts/xkcd_crawler.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    min_id: int = 1  # adjust here if necessary
+    max_id: int = get_xkcd()["num"]
+    print(f"Retrieving xkcd {min_id} through {max_id}...")
+
+    session = db.Session()
+    for n in tqdm(range(min_id, max_id)):
+        try:
+            xkcd = get_xkcd(n)
+            session.merge(db.XKCD(**xkcd))
+        except (api_requests.APIAccessError, TypeError):
```
Comment (RobinSikkens) [inline scripts/xkcd_crawler.py]: As the request is done with the `requests` library APIAccessError is unlikely to be thrown as it is a custom error for `api_requests`

**Review by RobinSikkens — APPROVED** [review]

File: devbot/commands/xkcd.py (hunk 2 above)
Comment (romanpeters) [reply devbot/commands/xkcd.py:24]: "https://xkcd.com/info.0.json" returns the most recent xkcd, not the first, so it already functions as desired.

**Pull request merged**